pkg/mesos/client.go

```diff
@@ -22,7 +22,29 @@ func NewClient(baseClient *httpclient.Client) *Client {
 	}
 }
 
-// Masters is returning the mesos masters of the connected cluster.
+// HTTP returns the HTTP client of the Mesos client.
+func (c *Client) HTTP() *httpclient.Client {
```

Comment:
I think we don't need this if we do a correct separation of concerns in the calling function.